fondant/manifest.py

```diff
@@ -188,12 +188,18 @@ def remove_subset(self, name: str) -> None:
 
         del self._specification["subsets"][name]
 
-    def evolve(self, component_spec: FondantComponentSpec) -> "Manifest":
-        """Evolve the manifest based on the provided component spec. The resulting manifest is
-        the expected result if the current manifest is provided to the component defined by the
-        component spec."""
+    def evolve(
+        self, metadata: dict, component_spec: FondantComponentSpec
+    ) -> "Manifest":
+        """Evolve the manifest based on the provided metadata and component spec. The resulting
+        manifest is the expected result if the current manifest is provided to the component defined
+        by the component spec."""
         evolved_manifest = self.copy()
 
+        # Update metadata
+        for key, value in metadata.items():
+            evolved_manifest.add_metadata(key, value)
```

Comment:
@RobbeSneyders I've added updating the `metadata` to the evolve method.

Reply:
Do we need to pass in the metadata? We should be able to get the `component_id` from the component spec, and I guess we can get the `run_id` from the input manifest.

I'ts just not clear to me yet how we can define the version of the component as well, since this depends on the commit hash of the component, not necessarily of the repo running the pipeline.

Reply:
Even without discussing if we need to pass in metadata to the component, we don't need to pass it to the `evolve` method here if we can infer it, right? The goal is that this method can be called from outside a component as well for eg. static analysis.

Reply:
Yes I've [removed](https://github.com/ml6team/fondant/pull/56/commits/cd0e23a7b4c5ff02a9436a12778318061d703959) `metadata` from the `evolve` method, it now only updates the component_id based on the name property of the component spec.